File: ConferenceManagement/web/src/main/webapp/src/app/shared/section.service.ts

import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Section} from "./section.model";
import {Observable} from "rxjs";
import {User} from "./user.model";
import {Proposal} from "./proposal.model";

@Injectable()
export class SectionService {
  private sectionUrl = 'http://localhost:8080/section';

  constructor(private httpClient: HttpClient) {
  }

  addSection(section: Section): Observable<Section>{
    console.log(section);
    return this.httpClient.post<Section>(this.sectionUrl + "/saveSection", section);
  }

  getSectionsForConference(conferenceID: number): Observable<Section[]> {
    return this.httpClient
      .post<Section[]>(this.sectionUrl + "/getSectionsForConference", conferenceID);
  }

  getSectionByID(sectionID: number): Observable<Section> {
    return this.httpClient
      .post<Section>(this.sectionUrl + "/getSectionByID", sectionID);
  }

  getCandidatesForSectionChair(conferenceID: number): Observable<User[]>{
    return this.httpClient.post<User[]>(this.sectionUrl + "/getCandidatesForSectionChair", conferenceID);
  }

  updateSectionProposals(section: Section): Observable<Section>{
    return this.httpClient.post<Section>(this.sectionUrl + "/updateSectionProposals", section);
  }

  getUnassignedAndAcceptedProposals(): Observable<Proposal[]>{
    return this.httpClient.get<Proposal[]>(this.sectionUrl + "/getUnassignedAndAcceptedProposals");
  }

  addParticipantInSection(sectionID: number, userID: number): Observable<void>{
    return this.httpClient.post<void>(this.sectionUrl + "/addParticipantInSection", [sectionID, userID]);
  }

  checkParticipantInSection(sectionID: number, userID: number): Observable<boolean>{
    return this.httpClient.post<boolean>(this.sectionUrl + "/checkParticipantInSection",[sectionID, userID]);
  }
}

File: ConferenceManagement/core/src/main/java/cms/core/service/SectionService.java
package cms.core.service;

import cms.core.domain.CMSUser;
import cms.core.domain.Proposal;
import cms.core.domain.Section;

import java.util.List;
import java.util.Optional;

public interface SectionService {

    Optional<Section> getSectionById(Long sectionId);

    List<Section> getSectionsByIDs(List<Long> sectionsIDs);

    List<Section> getAllByConferenceId(Long conferenceId);

    Section save(Section section);


    Optional<Section> updateSectionChair(Long sectionID, CMSUser futureChair);

    List<Proposal> getUnassignedAndAcceptedProposals();

    Section updateSectionProposals(Long sectionID, List<Long> proposalIDs);

    void addParticipantInSection(Long sectionID, Long userID);

    boolean checkParticipantInSection(Long sectionID, Long userID);

}
